spark/src/main/scala/filodb.spark/FiloRelation.scala

```diff
@@ -65,21 +65,38 @@ object FiloRelation extends StrictLogging {
     }.toSeq
   }
 
-  // Single partition query?
-  def singlePartitionQuery(projection: RichProjection,
-                           filterStuff: Seq[(Int, KeyType, Seq[Filter])]): Option[Any] = {
+  // Partition Query?
+  def partitionQuery(config: Config,
+                     projection: RichProjection,
+                     filterStuff: Seq[(Int, KeyType, Seq[Filter])]): Seq[Any] = {
+    val inqueryPartitionsLimit = config.getInt("columnstore.inquery-partitions-limit")
     // Are all the partition keys given in filters?
-    // Are all the filters EqualTo?
-    val equalPredValues: Seq[Any] = filterStuff.collect {
-      case (pos, keyType, Seq(EqualTo(_, equalValue))) => KeyFilter.parseSingleValue(keyType)(equalValue)
+    // Are the filters EqualTo, In?
+    val predicateValues = filterStuff.collect {
+      case (pos, keyType, Seq(EqualTo(_, equalValue))) =>
+        Set(KeyFilter.parseSingleValue(keyType)(equalValue))
+      case (pos, keyType, Seq(In(_, inValues))) =>
+        (KeyFilter.parseValues(keyType)(inValues.toSet))
     }
-    if (equalPredValues.length == projection.partitionColumns.length) {
-      if (equalPredValues.length == 1) Some(equalPredValues.head) else Some(equalPredValues)
+    logger.info(s"Push down predicate values: ${predicateValues.toString()}")
+    if (predicateValues.length == projection.partitionColumns.length) {
+      if(predicateValues.length == 1) {
+        predicateValues.flatten
+      }
+      else {
+        val predList = combine(predicateValues)
+        if (predList.size <= inqueryPartitionsLimit) predList else Nil
+      }
     } else {
-      None
+      Nil
     }
   }
 
+  def combine[A](xs: Traversable[Traversable[A]]): Seq[Seq[A]] =
```

Comment:
I don't quite understand what you're trying to do with combine.  Is this just to filter out the empty values?   Can't you just do `xs.filter(_.nonEmpty).toSeq` and get rid of the combine function?